beacon-chain/casper/validator.go

```diff
@@ -92,22 +92,18 @@ func SampleAttestersAndProposers(seed common.Hash, validators []*pb.ValidatorRec
 	return indices[:int(attesterCount)], indices[len(indices)-1], nil
 }
 
-// GetAttestersTotalDeposit from the pending attestations.
-func GetAttestersTotalDeposit(attestations []*pb.AggregatedAttestation) uint64 {
-	var numOfBits int
-	for _, attestation := range attestations {
-		numOfBits += int(shared.BitSetCount(attestation.AttesterBitfield))
-	}
-	// Assume there's no slashing condition, the following logic will change later phase.
-	return uint64(numOfBits) * params.DefaultBalance
+// AttestersTotalDeposit from the active attesters.
+func AttestersTotalDeposit(validatorIndices []uint32) uint64 {
+	return uint64(len(validatorIndices) * params.DefaultBalance)
 }
 
 // GetShardAndCommitteesForSlot returns the attester set of a given slot.
-func GetShardAndCommitteesForSlot(shardCommittees []*pb.ShardAndCommitteeArray, lcs uint64, slot uint64) (*pb.ShardAndCommitteeArray, error) {
-	if !(lcs <= slot && slot < lcs+params.CycleLength*2) {
-		return nil, fmt.Errorf("can not return attester set of given slot, input slot %v has to be in between %v and %v", slot, lcs, lcs+params.CycleLength*2)
+func GetShardAndCommitteesForSlot(shardCommittees []*pb.ShardAndCommitteeArray, lastStateRecalc uint64, slot uint64) (*pb.ShardAndCommitteeArray, error) {
+	if !(lastStateRecalc <= slot && slot < lastStateRecalc+params.CycleLength*2) {
```

Comment:
Shouldnt this be `slot < lastStateRecalc+params.CycleLength` ?

Reply:
The range is actually 2 * cycle length same for recent block hashes:

```
def get_shards_and_committees_for_slot(crystallized_state, slot):
    ifh_start = crystallized_state.last_state_recalc - CYCLE_LENGTH
    assert ifh_start <= slot < ifh_start + CYCLE_LENGTH * 2
    return crystallized_state.shard_and_committee_for_slots[slot - ifh_start]
```